save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/event/VulnerabilityMetadataListener.kt

```diff
@@ -0,0 +1,46 @@
+package com.saveourtool.save.backend.controllers.vulnerability.event
+
+import com.saveourtool.save.backend.service.NotificationService
+import com.saveourtool.save.backend.service.UserDetailsService
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.entities.Notification
+import com.saveourtool.save.entities.cosv.VulnerabilityMetadata
+import com.saveourtool.save.entities.vulnerability.VulnerabilityStatus
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+/**
+ * A vulnerability metadata listener for sending notifications.
+ */
+@Component
+class VulnerabilityMetadataListener(
+    private val userDetailsService: UserDetailsService,
+    private val notificationService: NotificationService,
+) {
+    /**
+     * @param vulnerability new VulnerabilityMetadata
+     */
+    @EventListener
+    fun createVulnerabilityMetadata(vulnerability: VulnerabilityMetadata) {
```

Comment:
the only thing I am worried here is about the argument type. I would have made some Event class for it, it can be a value/inline class, for example

Reply:
the same for User, after it we checks the vulnerability status. I guess we create create a required event in place when `applicationEventPublisher` is called

Reply:
Added Event class